cookbook/persistance/key-value.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: page
+title: "Storing key-value data on disk"
+permalink: /cookbook/persistence/key-value/
+---
+
+If you have a relatively small collection of key-values that you'd like to save, you should use the [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) APIs
+
+Normally we would have to write native platform integration for storing data for both platforms, but fortunetly [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin can be used for that. 
+
+Shared preferences plugin wraps `NSUserDefaults` (on iOS) and `SharedPreferences` (on Android), providing a persistent store for simple data. Data is persisted to disk automatically and asynchronously.
+
+## Setup
+
+Before you start, you need to add [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin to you `pubspec.yaml` file:
+
+```yaml
+dependencies:
+  flutter:
+    sdk: flutter
+  shared_preferences: "<newest version>"
+```
+
+## Saving, reading and removing data
+
+In order to access store simply use `SharedPreferences` class.
+
+```dart
+// import shared preferences plugin
+import 'package:shared_preferences/shared_preferences.dart';
+
+// obtain shared preferences 
+SharedPreferences prefs = await SharedPreferences.getInstance();
+
+// get  value by key, return null if not available
+int counter = (prefs.getInt('counter') ?? 0) + 1;
+print('Pressed $counter times.');
+
+// set new value
+prefs.setInt('counter', counter);
+
+// removes value for given key
+prefs.remove('counter');
+
+```
+
+In example above we load data from `counter` key, if it not exists, `0` is returned. 
+
+For saving data we call `set` method. Note that data is persisted on disk automatically and asynchronously. If you want to get notified when data is saved use `commit()` function. 
+
+Adequately setter and getter methods are available for all primitive types.
```

Comment:
"Adequately setter and getter methods are available for all primitive types." -- Please remove "Adequately" at the beginning of this sentence.